save-cloud-common/src/nativeMain/kotlin/com/saveourtool/save/utils/FileUtils.kt

```diff
@@ -46,6 +48,17 @@ fun FileSystem.createAndWriteIfNeeded(fileName: String?, lines: List<String>?) =
     write(path, true) { lines?.forEach { codeLine -> writeUtf8("$codeLine\n") } }
 }
 
+/**
+ * Mark all files in [dir] as executable.
+ *
+ * Note that files from subdirectories are not marked as executable.
+ *
+ * @param dir [Path] to directory
+ */
+fun markAllFilesAsExecutable(dir: Path) {
```

Comment:
it could be probably an extension method
also you should be sure, that `dir` is really directory

Reply:
Moved existing `unzipIfRequired` to `save-cloud-common` and refactored it a bit